File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size();
        int ans=0;
        int i=0,j=0;
        unordered_set<char> st;
        for(int i=0;i<n;i++){
            if(st.find(s[i])==st.end()){
                st.insert(s[i]);
            }
            else{
                while(j<n&&j<i &&st.find(s[i])!=st.end()){
                    st.erase(s[j]);
                    j++;
                }
                st.insert(s[i]);
            }
            ans=max(ans,i-j+1);
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let n=s.length;
    let right=0,left=0,ans=0;
    const frq=new Set();
    while(right<n){
        while(frq.has(s[right])){
            frq.delete(s[left++]);
        }
        ans=Math.max(ans,right-left+1);
        frq.add(s[right++]);
    }
    return ans;

};